File: STRINGS/2587/main.js
function isPossible(word1, word2, word3) {
    let first = -1;
    let second = -1;
    for (let index in word3)
        if (word3[index] == '_')
            first < 0 ? first = index : second = index;
    return word1[first] == word2[second] || word1[second] == word2[first];
}


let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n');
let c = parseInt(lines.shift());
while (c--) {
    let word1 = lines.shift();
    let word2 = lines.shift();
    let word3 = lines.shift();
    console.log(isPossible(word1, word2, word3) ? 'Y' : 'N');
}

File: STRINGS/2587/main.dart
import 'dart:io';

bool isPossible(String word1, String word2, String word3) {
  int first = -1;
  int second = -1;
  for (int index = 0; index < word3.length; index++) {
    if (word3[index].contains('_')) {
      first < 0 ? first = index : second = index;
    }
  }
  return word1[first] == word2[second] || word1[second] == word2[first];
}

void main() {
  int c = int.parse(stdin.readLineSync()!);
  while (c-- > 0) {
    String word1 = stdin.readLineSync()!;
    String word2 = stdin.readLineSync()!;
    String word3 = stdin.readLineSync()!;
    print(isPossible(word1, word2, word3) ? 'Y' : 'N');
  }
}
